Intermittent test failure in BrokerClientIntegrationTest.testUnsupportedBatchMessageConsumer

```
2019-02-07\T\04:39:49.533 [ERROR] Tests run: 14, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 54.155 s <<< FAILURE! - in org.apache.pulsar.client.impl.BrokerClientIntegrationTest
2019-02-07\T\04:39:49.533 [ERROR] testUnsupportedBatchMessageConsumer(org.apache.pulsar.client.impl.BrokerClientIntegrationTest)  Time elapsed: 2.131 s  <<< FAILURE!
org.apache.pulsar.client.api.PulsarClientException$ConsumerBusyException: Exclusive consumer is already connected
	at org.apache.pulsar.client.impl.ClientCnx.getPulsarClientException(ClientCnx.java:825)
	at org.apache.pulsar.client.impl.ClientCnx.handleError(ClientCnx.java:536)
	at org.apache.pulsar.common.api.PulsarDecoder.channelRead(PulsarDecoder.java:157)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1414)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:945)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:806)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:404)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:304)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/client/impl/BrokerClientIntegrationTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/client/impl/BrokerClientIntegrationTest.java
@@ -310,15 +310,23 @@ public class BrokerClientIntegrationTest extends ProducerConsumerBase {
         final String topicName = "persistent://my-property/my-ns/my-topic1";
         final String subscriptionName = "my-subscriber-name" + subType;
 
-        ConsumerImpl<byte[]> consumer1 = (ConsumerImpl<byte[]>) pulsarClient.newConsumer().topic(topicName)
-                .subscriptionName(subscriptionName).subscriptionType(subType).subscribe();
+        ConsumerImpl<byte[]> consumer1 = (ConsumerImpl<byte[]>) pulsarClient.newConsumer()
+                .topic(topicName)
+                .subscriptionName(subscriptionName)
+                .subscriptionType(subType)
+                .subscribe();
 
         final int numMessagesPerBatch = 10;
 
-        Producer<byte[]> producer = pulsarClient.newProducer().topic(topicName).create();
-        Producer<byte[]> batchProducer = pulsarClient.newProducer().topic(topicName).enableBatching(true)
+        Producer<byte[]> producer = pulsarClient.newProducer()
+                    .topic(topicName)
+                    .enableBatching(false)
+                    .create();
+        Producer<byte[]> batchProducer = pulsarClient.newProducer()
+                .topic(topicName).enableBatching(true)
                 .batchingMaxPublishDelay(Long.MAX_VALUE, TimeUnit.SECONDS)
-                .batchingMaxMessages(numMessagesPerBatch).create();
+                .batchingMaxMessages(numMessagesPerBatch)
+                .create();
 
         // update consumer's version to incompatible batch-message version = Version.V3
         Topic topic = pulsar.getBrokerService().getOrCreateTopic(topicName).get();
@@ -357,12 +365,15 @@ public class BrokerClientIntegrationTest extends ProducerConsumerBase {
         batchProducer.flush();
 
         // consumer should have not received any message as it should have been disconnected
-        msg = consumer1.receive(2, TimeUnit.SECONDS);
+        msg = consumer1.receive(100, TimeUnit.MILLISECONDS);
         assertNull(msg);
 
         // subscribe consumer2 with supporting batch version
         PulsarClient newPulsarClient = newPulsarClient(lookupUrl.toString(), 0); // Creates new client connection
-        Consumer<byte[]> consumer2 = newPulsarClient.newConsumer().topic(topicName).subscriptionName(subscriptionName)
+        Consumer<byte[]> consumer2 = newPulsarClient.newConsumer()
+                .topic(topicName)
+                .subscriptionName(subscriptionName)
+                .subscriptionType(subType)
                 .subscribe();
 
         messageSet.clear();
@@ -751,7 +762,7 @@ public class BrokerClientIntegrationTest extends ProducerConsumerBase {
     /**
      * It verifies that if broker fails to complete producer/consumer operation then client times out rather waiting
      * forever.
-     * 
+     *
      * @throws PulsarClientException
      */
     @Test(expectedExceptions = PulsarClientException.TimeoutException.class, timeOut = 10000)
@@ -832,5 +843,5 @@ public class BrokerClientIntegrationTest extends ProducerConsumerBase {
         producer.close();
         consumer.close();
     }
-    
+
 }

